staging/src/k8s.io/cluster-bootstrap/token/util/helpers.go

```diff
@@ -131,3 +134,80 @@ func ValidateUsages(usages []string) error {
 	}
 	return nil
 }
+
+// GetSecretString returns the string value for the given key in the specified Secret
+// If there is an error or if the key doesn't exist, an empty string is returned.
+func GetSecretString(secret *v1.Secret, key string) string {
+	if secret.Data == nil {
+		return ""
+	}
+	if val, ok := secret.Data[key]; ok {
+		return string(val)
+	}
+	return ""
+}
+
+// IsSecretExpired will identify whether the secret expires
+func IsSecretExpired(secret *v1.Secret) bool {
```

Comment:
Wouldn't it make sense to factor this as a function that just extracts the expiration time + a function that compares with now, ideally involving a clock so that now can be faked and tested?